### PR TITLE
feat(AutoResponse): add `discordJSVersions` property

### PR DESCRIPTION
**Pull request by Syjalo**
This PR adds `AutoResponse#discordJSVersions` to send responses based on the version

### CODE REVIEW

File: src/events/messageCreate.ts [hunk 1]
```diff
@@ -25,17 +32,37 @@ export default class implements Event {
...
+						supportChannels.push(V14_SUPPORT_CHANNEL);
+						questionsForumTags.push(...V14_QUESTIONS_FORUM_TAGS);
+					}
+
+					if (
+						(!supportChannels.includes(message.channel.id) &&
+							(message.channel.type !== ChannelType.PublicThread ||
+								!message.channel.appliedTags.some((tag) => tag.length && questionsForumTags.includes(tag)))) ||
+						!response.keyphrases.some((phrase) => phrase.length && message.content.toLowerCase().includes(phrase))
+					) {
```
Comment (Syjalo) [inline src/events/messageCreate.ts:53]: Not so cute. I can replace with variables.

File: src/events/messageCreate.ts [hunk 2]
```diff
@@ -25,17 +32,37 @@ export default class implements Event {
...
+
+					if (response.discordJSVersions.includes(DiscordJSVersion.V13)) {
+						supportChannels.push(V13_SUPPORT_CHANNEL);
+						questionsForumTags.push(...V13_QUESTIONS_FORUM_TAGS);
+					}
+
+					if (response.discordJSVersions.includes(DiscordJSVersion.V14)) {
+						supportChannels.push(V14_SUPPORT_CHANNEL);
+						questionsForumTags.push(...V14_QUESTIONS_FORUM_TAGS);
+					}
```
Comment (jaw0r3k) [inline src/events/messageCreate.ts:46]: It looks a little bit unnecessary

Comment (Syjalo) [reply src/events/messageCreate.ts:46]: What do you suggest?

**Comment by almostSouji** [comment]
i understand the idea, but i think I prefer discerning them based on the error or having both with a short line each, should it not be discernable

**Comment by Syjalo** [comment]
But what if an autoresponse applies only to one version? Like `<Guild>.members.me`

**Pull request closed without merging**